### PR TITLE
WIP: Assign product image to selected combination in bulk action

### PR DESCRIPTION
**Pull request by fatmaBouchekoua**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | In bulk actions, we must be able to select the images to be associated with the chosen combinations. When you click on "select images", it displays a drop-down menu including: A checkbox, the thumbnail of the image and the label of the image.
| Type?         | new feature
| Category?     | BO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | http://forge.prestashop.com/browse/FF-74
| How to test?  | Edit a product with combination from the BO and try to assign images to some combinations.

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/7890)

### CODE REVIEW

**Comment by vincentbz** [comment]
Thanks Fatma, it works well :)

There are just a few UI feedbacks : 
 - could you use the checkbox used for checking the combinations
 - and the font used for combination names


image

**Comment by fatmaBouchekoua** [comment]
Hello @vincentbz ,

I've used the same as indicated in Zeplin. 
@Leadesign what do you think?

File: admin-dev/themes/new-theme/scss/pages/_product_page.scss [hunk 1]
```diff
@@ -1050,3 +1050,78 @@
...
 }
+
+#select2-product_combination_bulk_images-results .select2-results__option--highlighted[aria-selected],
+#select2-product_combination_bulk_images-results .select2-results__option[aria-selected=true] {
+  background-color: inherit;
+  color: inherit;
+}
+
+#select2-product_combination_bulk_images-results li.select2-results__option[aria-selected="true"]::before {
+  background: #3ED2F0;
```
Comment (maximebiloe) [inline admin-dev/themes/new-theme/scss/pages/_product_page.scss]: We CSS variables, please don't use directly the hexadecimal code for colors

File: admin-dev/themes/new-theme/scss/pages/_product_page.scss [hunk 2]
```diff
@@ -1050,3 +1050,78 @@
...
+  border-color: #3ED2F0;
+  content: '\E5CA';
+  font-family: Material Icons;
+  color: white;
+  line-height: rem(12px);
```
Comment (maximebiloe) [inline admin-dev/themes/new-theme/scss/pages/_product_page.scss]: rem() is now deprecate, use directly the rem value

**Review by maximebiloe — CHANGES_REQUESTED** [review]

**Comment by maximebiloe** [comment]
After everything has been fixed, please rebase, you have more than 700 commits behind

File: admin-dev/themes/new-theme/scss/pages/_product_page.scss (hunk 1 above)
Comment (fatmaBouchekoua) [reply admin-dev/themes/new-theme/scss/pages/_product_page.scss]: OK done for the ones I found in the PrestaKit

**Comment by fatmaBouchekoua** [comment]
@maximebiloe I've rebased it.

**Comment by fatmaBouchekoua** [comment]
Hello @marionf ,
The fields are well aligned on my shop.
I've fixed the other problems. Can you retest please?
Thanks !

**Comment by marionf** [comment]
Hello @fatmaBouchekoua 

The fields are not aligned.
Also, I have 2 checkboxes and the font in the drop down should be the same as used for combination names and defined in the UI-Kit. The color of the font should be #363A41 as defined in the UI-Kit.

When I refresh the page, I didn't see anymore the images in the drop down and have "No results found".

pr7890-3

**Comment by fatmaBouchekoua** [comment]
Hello @marionf ,
The assets file was not well compiled, so that you get 2 checkbox and not the requested font color.
As I said, the fields are well aligned on my shop, try to clear cache.
Also I refresh the page and the images still exist.

capture 2017-08-04 at 15 21 31

**Comment by marionf** [comment]
Hello @fatmaBouchekoua 
It's ok for the fields they are well aligned and it's ok for the font also.
But I still have the problem with the empty image dropdown when I refresh the page.
I tested with @vincentbz & @prasanthSelvar, they have also this problem.

**Comment by fatmaBouchekoua** [comment]
Hello @marionf ,
This problem occurs when you select combination. 
I've made changes! Can you recheck please?

Sorry for testing this PR many times.
Thanks!

**Comment by marionf** [comment]
Hello @fatmaBouchekoua,
It's ok the dropdown isn't empty anymore.
I will retest after the code review because changes are requested.

File: admin-dev/themes/new-theme/js/product-page/product-bulk-combinations.js [hunk 3]
```diff
@@ -202,8 +303,22 @@ class Combination {
   updateForm(values) {
     values.forEach((valueObject) => {
       var valueId = valueObject.id.substr(this.inputBulkPattern.length);
-      $('#'+this.convertInput(valueId)).val(valueObject.value);
-    });
+    if (this.convertInput(valueId) == this.inputPattern + 'id_image_attr') {
```
Comment (maximebiloe) [inline admin-dev/themes/new-theme/js/product-page/product-bulk-combinations.js]: Still misindented

File: admin-dev/themes/new-theme/scss/pages/_product_page.scss [hunk 4]
```diff
@@ -1065,3 +1065,63 @@
...
+  color: $gray-medium;
+  margin: 0.6429rem 0 0 0.6429rem;
+}
+
+#select2-product_combination_bulk_images-results li.select2-results__option input[type=checkbox] {
+  margin: 0 0.625rem 0 0.313rem;
+  vertical-align: middle;
+}
+
+#select2-product_combination_bulk_images-results::-webkit-scrollbar {
```
Comment (maximebiloe) [inline admin-dev/themes/new-theme/scss/pages/_product_page.scss]: Why are you using the browser prefixes? It is automatically added with webpack...

File: admin-dev/themes/new-theme/scss/pages/_product_page.scss [hunk 5]
```diff
@@ -1065,3 +1065,63 @@
...
+  border-radius: 7.143rem;
+  background-color: #bbcdd2;
+}
+
+#select2-product_combination_bulk_images-container .select2-container--prestakit .select2-selection--multiple {
+  padding: inherit;
+  padding-right: 1.429rem;
+}
+
+#select2-product_combination_bulk_images-container .select2-container--prestakit .select2-selection--multiple::before {
```
Comment (maximebiloe) [inline admin-dev/themes/new-theme/scss/pages/_product_page.scss]: I don't think you're using advantages of sass here...

File: src/PrestaShopBundle/Controller/Admin/CombinationController.php [hunk 6]
```diff
@@ -108,4 +108,29 @@ public function getProductCombinationsAction($idProduct)
...
+     * get product images to assigning for combination
+     * @param $idProduct
+     * @return \Symfony\Component\HttpFoundation\JsonResponse
+     * @throws \Exception
+     */
+    public function getProductImagesCombinationsAction($idProduct)
+    {
+        $response = new JsonResponse();
+        $product = new \Product($idProduct);
+        $results = [];
```
Comment (maximebiloe) [inline src/PrestaShopBundle/Controller/Admin/CombinationController.php]: no short tag for array

File: src/PrestaShopBundle/Controller/Admin/CombinationController.php [hunk 7]
```diff
@@ -108,4 +108,29 @@ public function getProductCombinationsAction($idProduct)
...
+    /**
+     * get product images to assigning for combination
+     * @param $idProduct
+     * @return \Symfony\Component\HttpFoundation\JsonResponse
+     * @throws \Exception
+     */
+    public function getProductImagesCombinationsAction($idProduct)
+    {
+        $response = new JsonResponse();
+        $product = new \Product($idProduct);
```
Comment (maximebiloe) [inline src/PrestaShopBundle/Controller/Admin/CombinationController.php]: No call of legacy code in the new one

File: src/PrestaShopBundle/Controller/Admin/CombinationController.php [hunk 8]
```diff
@@ -108,4 +108,29 @@ public function getProductCombinationsAction($idProduct)
...
+        $results = [];
+        $context = \Context::getContext();
```
Comment (maximebiloe) [inline src/PrestaShopBundle/Controller/Admin/CombinationController.php]: No call of legacy code in the new one

**Review by maximebiloe — CHANGES_REQUESTED** [review]

File: admin-dev/themes/new-theme/scss/pages/_product_page.scss (hunk 4 above)
Comment (fatmaBouchekoua) [reply admin-dev/themes/new-theme/scss/pages/_product_page.scss]: I think that this webpack autofixer version does not handle this for scrollbar selector. 
https://github.com/postcss/autoprefixer/issues/821

File: src/PrestaShopBundle/Form/Admin/Product/ProductCombinationBulk.php [hunk 9]
```diff
@@ -94,6 +94,11 @@ public function buildForm(FormBuilderInterface $builder, array $options)
             ->add('minimal_quantity', 'Symfony\Component\Form\Extension\Core\Type\NumberType', array(
                 'required' => false,
                 'label' => $this->translator->trans('Minimum quantity', [], 'Admin.Catalog.Feature'),
+            ))
+            ->add('images', 'Symfony\Component\Form\Extension\Core\Type\ChoiceType', array(
+                'required' => false,
+                'multiple' => true,
+                'label' => $this->translator->trans('Image', [], 'Admin.Catalog.Feature'),
```
Comment (AlexEven) [inline src/PrestaShopBundle/Form/Admin/Product/ProductCombinationBulk.php]: Can you please use "Admin.Global" as a domain? Thanks!

Comment (fatmaBouchekoua) [reply src/PrestaShopBundle/Form/Admin/Product/ProductCombinationBulk.php]: Done!

File: admin-dev/themes/new-theme/js/product-page/product-bulk-combinations.js [hunk 10]
```diff
@@ -76,9 +111,30 @@ export default function() {
...
+      });
+      $('#product_combination_bulk_images').on("select2:select", function () {
+        $('#product_combination_bulk_images').select2('open');
+        that.setBulkActionImagesCheckboxStatus();
+      });
+      $('#product_combination_bulk_images').on("select2:unselect", function () {
+        $('#product_combination_bulk_images').select2('open');
+        that.setBulkActionImagesCheckboxStatus();
+      });
+      $('body').on('DOMNodeInserted', function () {
```
Comment (eternoendless) [inline admin-dev/themes/new-theme/js/product-page/product-bulk-combinations.js:131]: Why this?

File: admin-dev/themes/new-theme/js/product-page/product-bulk-combinations.js [hunk 11]
```diff
@@ -202,8 +303,22 @@ class Combination {
...
       var valueId = valueObject.id.substr(this.inputBulkPattern.length);
-      $('#'+this.convertInput(valueId)).val(valueObject.value);
-    });
+      if (this.convertInput(valueId) == this.inputPattern + 'id_image_attr') {
+        if (valueObject.value != null) {
+          var combinationId = this.domId
+          valueObject.value.forEach(function (element) {
+            $('#combination_form_' + combinationId + ' .product-combination-image').each(function () {
+              if ($(this).find('input').val() == element) {
+                $(this).find('input').prop('checked', true);
```
Comment (eternoendless) [inline admin-dev/themes/new-theme/js/product-page/product-bulk-combinations.js:312]: `$(this).find('input')` is being performed twice here

Comment (eternoendless) [reply admin-dev/themes/new-theme/js/product-page/product-bulk-combinations.js:312]: You could also store `$(this)` in a variable

File: admin-dev/themes/new-theme/scss/pages/_product_page.scss [hunk 12]
```diff
@@ -1065,3 +1065,63 @@
...
+    background-color: $gray-light;
+  }
+
+}
+
+#select2-product_combination_bulk_images-container .select2-container--prestakit .select2-selection--multiple {
+  padding: inherit;
+  padding-right: 1.429rem;
+
+  &::before {
```
Comment (nihco2) [inline admin-dev/themes/new-theme/scss/pages/_product_page.scss:1115]: Hi @fatmaBouchekoua, I think you could use the new version of UI-Kit without override select2 input style: https://www.npmjs.com/package/prestakit

File: admin-dev/themes/new-theme/scss/pages/_product_page.scss [hunk 13]
```diff
@@ -1065,3 +1065,63 @@
...
+    position: absolute;
+    top: 50%;
+    right: 0.3571rem;
+    transform: translateY(-50%);
+  }
+
+  ul {
+    display: none;
+  }
+}
```
Comment (nihco2) [inline admin-dev/themes/new-theme/scss/pages/_product_page.scss:1127]: It seems the products label are missing:
capture d ecran 2017-09-25 a 10 44 03

**Comment by eternoendless** [comment]
Closing this due to inactivity. It should be redone following #12773

**Pull request closed without merging**